ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/container/v1/container/clientlibs/editor/js/editDialog.js

```diff
@@ -42,6 +42,259 @@
+        hideContainersExcept(schemaTypeSelected);
+        if (schemaTypeSelected == SCHEMA) {
+            if (lastSelectedXsd) {
+                $(SCHEMA_ROOT_ELEMENT_SELECTOR).show();
+                createRootElementSelectionMarkup(lastSelectedXsd);
+            }
+        } else {
+            removeXsdRootElementMarkup();
+        }
+    };
+
+    function prefillSchema(schemaType, dialog){
+        var schemaRef = dialog.find(SCHEMA_REF)[0].value,
+            xsdRef = dialog.find(XSD_REF)[0].value;
+        if (schemaType == SCHEMA) {
+            $(SCHEMA_DROPDOWN_SELECTOR).val(schemaRef ? schemaRef : xsdRef);
+        } else if (schemaType == FORM_DATA_MODEL) {
+            $(FDM_DROPDOWN_SELECTOR).val(schemaRef ? schemaRef : xsdRef);
+        }
+        if(xsdRef){
+            fillXsdDetails(xsdRef);
+        }
+    };
+
+    function schemaSelectorOnChanged(dialog) {
+        var selectedSchema = dialog.find(SCHEMA_DROPDOWN_SELECTOR)[0].value;
+        if (selectedSchema.indexOf(XSD_EXTENSION) != -1) {
+            $(SCHEMA_ROOT_ELEMENT_SELECTOR).show();
+            dialog.find(XSD_REF)[0].value = selectedSchema;
+            dialog.find(SCHEMA_REF)[0].value = '';
+            createRootElementSelectionMarkup(selectedSchema);
+        } else if (selectedSchema.indexOf(JSON_EXTENSION) != -1) {
+            $(SCHEMA_ROOT_ELEMENT_SELECTOR).hide();
+            removeXsdRootElementMarkup();
+            handleJsonSelection(selectedSchema);
+            dialog.find(SCHEMA_REF)[0].value = selectedSchema;
+            dialog.find(XSD_REF)[0].value = '';
+        } else {
+            dialog.find(SCHEMA_REF)[0].value = selectedSchema;
+            dialog.find(XSD_REF)[0].value = '';
+        }
+        if (configuredFormModel) {
+            confirmFormModelChange(selectedSchema, $(SCHEMA_DROPDOWN_SELECTOR));
+        } else {
+            toBeConfiguredFormModel = selectedSchema;
+        }
+    };
+
+    function fdmSelectorOnChanged(dialog) {
+        var selectedSchema = dialog.find(FDM_DROPDOWN_SELECTOR)[0].value;
+        dialog.find(SCHEMA_REF)[0].value = selectedSchema;
+        dialog.find(XSD_REF)[0].value = '';
+        if (configuredFormModel) {
+            confirmFormModelChange(selectedSchema, $(FDM_DROPDOWN_SELECTOR));
+        } else {
+            toBeConfiguredFormModel = selectedSchema;
+        }
+    };
+
+    function handleJsonSelection(selectedSchema) {
+        lastSelectedXsd = '';
+        $(SCHEMA_DROPDOWN_SELECTOR).val(selectedSchema);
+        removeXsdRootElementMarkup();
+    };
+
+    function createRootElementSelectionMarkup(selectedSchema, preselectedRootElement) {
+        lastSelectedXsd = selectedSchema;
+        var path = selectedSchema + '/jcr:content/renditions/original';
+        $.ajax({
+            type: "POST",
+            url: Granite.HTTP.externalize("/libs/fd/fm/gui/content/forms/guide/validate.html?xsdFilePath=" + encodeURIComponent(path) + "&assetType=guide")
```

Comment:
is this pubic API? we cannot use internal HTTP APIs here. If its intended to be a public API, then it should follow the API guidelines (currently this end point doesn't).

Reply:
removed it, as not needed without xsd